sql/core/src/test/resources/sql-tests/results/array.sql.out

```diff
@@ -431,6 +431,104 @@ struct<get(array(1, 2, 3), -1):int>
 NULL
 
 
+-- !query
+select array_insert(array(1, 2, 3), 3, 4)
+-- !query schema
+struct<array_insert(array(1, 2, 3), 3, 4):array<int>>
+-- !query output
+[1,2,4,3]
+
+
+-- !query
+select array_insert(array(2, 3, 4), 0, 1)
```

Comment:
pos=0 also means the first item, is this expected? If so, I think we need to document it.

Reply:
No problem at all :)

For pos=0, this was to implement the behaviour specified [here](https://github.com/apache/spark/pull/38867#discussion_r1041949049). However, the other spark functions (elementAt, slice) throw an exception i.e. see [slice](https://github.com/apache/spark/blob/c13fea90595cb1489046135c45c92d3bacb85818/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala#L1662). What would your preference be? Maybe we should throw an exception, considering we are no longer implementing the snowflake behaviour?

Reply:
CC @LuciferYang for opinion also, as original commenter, and @cloud-fan for any additional context

Reply:
Sorry, missed some message

Reply:
No worries @LuciferYang - it was about this:

https://github.com/apache/spark/pull/38867#discussion_r1041949049

We have implemented array_insert(arr, 0, item) as array_prepend; let me know if this is incorrect

Reply:
I think this 0 behavior is a trap. It invites the belief that array_isert is 0 based.